x/emissions/module/rewards/rewards.go

```
@@ -152,29 +149,23 @@ func EmitRewards(args EmitRewardsArgs) error {
 			continue
 		}
 
-		// Add rewardInTopicToReputers to totalRewardToStakedReputers
-		totalRewardToStakedReputers, err = totalRewardToStakedReputers.Add(rewardInTopicToReputers)
+		// Add to monthly reputer rewards
+		rewardInTopicToReputersInt, err := rewardInTopicToReputers.SdkIntTrim()
 		if err != nil {
-			return errors.Wrapf(
-				err,
-				"Error finding sum of rewards to Reputers: totalReward: %s , rewardInTopic: %s",
-				totalRewardToStakedReputers.String(),
-				rewardInTopicToReputers.String(),
-			)
+			return errors.Wrapf(err, "failed to convert reward in topic to reputers to int")
 		}
-	}
-
-	// Log and handle the final totalRewardToStakedReputers
-	Logger(args.Ctx).Debug("Paid out to staked reputers", "totalRewardToStakedReputers", totalRewardToStakedReputers.String(), "topics", len(topicRewards))
-
-	if !totalRewardTreasury.IsZero() && uint64(args.BlockHeight)%args.ModuleParams.BlocksPerMonth == 0 {
-		percentageToStakedReputers, err := totalRewardToStakedReputers.Quo(totalRewardTreasury)
+		err = args.K.AddMonthlyReputerRewards(args.Ctx, rewardInTopicToReputersInt)
+		if err != nil {
+			return errors.Wrapf(err, "failed to add monthly reputer rewards")
+		}
+		// Add to monthly topic rewards
+		topicRewardInt, err := topicReward.SdkIntTrim()
 		if err != nil {
-			return errors.Wrapf(err, "failed to calculate percentage to staked reputers")
+			return errors.Wrapf(err, "failed to convert topic reward to int")
 		}
-		err = args.K.SetPreviousPercentageRewardToStakedReputers(args.Ctx, percentageToStakedReputers)
+		err = args.K.AddMonthlyTopicRewards(args.Ctx, topicRewardInt)
```

Small thing here, but wouldn't it be better to just store and add at the end? For efficiency (one write to the store).
  
Also makes me think the "partial modification" state that it will be left if the logic fails in some rewardableTopic, preventing others from being processed. 
1. Should those be a `continue`, and in that case, it shouldn't do any modification?
2. If it still should return an error, it that partial state change correct?

---

I think we need to start thinking about where a `panic(err)` is actually desirable — anything related to tokenomics is a strong candidate for that. 

Some anomalies can happen that I'm sure would be better to just not move forward if they happen.

---

This is usually the way it is managed, and this doesn't require a panic but just returning the error in the end blocker instead of silencing it.

I'd tend to prefer dealing with chain halts than state inconsistencies, the latter could be a nightmare on the long term.

To mitigate this, a lot of errors can occur on unhealthy nodes with difficulties reading/writing to the database, in such cases the node should stop

---

This is usually the way it is managed, and this doesn't require a panic but just returning the error in the end blocker instead of silencing it.

I'd tend to prefer dealing with chain halts than state inconsistencies, the latter could be a nightmare on the long term.

To mitigate this, a lot of errors can occur on unhealthy nodes with difficulties reading/writing to the database, in such cases the node should stop

---

+1 on reviewing the usage of panic and make it consistent through the codebase.